web/src/compose_ui.js

```diff
@@ -500,3 +511,49 @@ export function get_submit_button() {
     }
     return $("#compose-send-button");
 }
+
+export function on_compose_select_stream_update(new_value) {
+    const $stream_header_colorblock = $("#compose_stream_selection_dropdown").find(
+        ".stream_header_colorblock",
+    );
+    stream_bar.decorate(new_value, $stream_header_colorblock, true);
+    compose.update_on_recipient_change();
+    $("#stream_message_recipient_topic").trigger("focus").trigger("select");
+}
+
+export let compose_stream_widget;
+export function initialize_compose_stream_dropdown() {
+    const streams_list = stream_data
+        .subscribed_subs()
+        .filter((stream) => stream_data.can_post_messages_in_stream(stream))
+        .map((stream) => ({
+            name: stream.name,
+            value: stream.name,
```

Comment:
You can update this object to add `stream` field to add the stream icon to the dropdown to make it consistent with othe places. You can check code in `settings_org.init_dropdown_widgets` for reference.

Reply:
Oh cool!! Thanks :)

Reply:
Switched to this in the most recent push, way cleaner :) Thanks for pointing it out!